Publisher-and-Subscriber-Controls/js/app.js

```diff
@@ -0,0 +1,172 @@
+  }
+}
+
+function toggleStyle(button) { // toggles button css class
+  button.classList.toggle('toggle-button-on');
+  button.classList.toggle('toggle-button-off');
+}
+
+function toggleMicrophone() {
+  if (publisher.session) publisher.publishAudio(!publisher.stream.hasAudio);
+  const microphoneToggle = document.getElementById('microphone-toggle');
+  toggleStyle(microphoneToggle);
+}
+
+function toggleCamera() {
+  if (publisher.session) publisher.publishVideo(!publisher.stream.hasVideo);
+  const cameraToggle = document.getElementById('camera-toggle');
+  toggleStyle(cameraToggle);
+}
+
+function toggleAudio() {
+  subscribedToAudio = !subscribedToAudio;
+  if (subscriber && subscriber.session) subscriber.subscribeToAudio(subscribedToAudio);
+  const audioToggle = document.getElementById('audio-toggle');
+  toggleStyle(audioToggle);
+}
+
+function toggleVideo() {
+  subscribedToVideo = !subscribedToVideo;
+  if (subscriber && subscriber.session) subscriber.subscribeToVideo(subscribedToVideo);
+  const videoToggle = document.getElementById('video-toggle');
+  toggleStyle(videoToggle);
+}
+
+function setButtons(on, buttons) {
+  buttons.forEach(button => {
+    if (on) {
+      button.classList.remove('toggle-button-off');
+      button.classList.add('toggle-button-on');
+    } else {
+      button.classList.remove('toggle-button-on');
+      button.classList.add('toggle-button-off');
+    }
+  });
+}
+
+function initializeSession() {
+  const session = OT.initSession(apiKey, sessionId);
+
+  // Subscribe to a newly created stream
+  session.on('streamCreated', function subscribe(event) {
+    subscriber = session.subscribe(event.stream, 'subscriber', {
+      insertMode: 'append',
+      width: '100%',
+      height: '100%'
+    }, handleError);
+    if (subscriber && subscriber.session) {
+      subscribedToAudio = subscriber.stream.hasAudio;
+      subscribedToVideo = subscriber.stream.hasVideo;
+      console.log('set');
+      setButtons(subscribedToAudio, [document.getElementById('audio-toggle')]);
+      setButtons(subscribedToVideo, [document.getElementById('video-toggle')]);
+    }
+  });
+
+  // Create a publisher
+  publisher = OT.initPublisher('publisher', {
+    insertMode: 'append',
+    width: '100%',
+    height: '100%'
+  }, handleError);
+
+  // If you disconnect from a call, overrride the default behaviour.
+  publisher.on('streamDestroyed', function preventSessionDefault(event) {
+    if (session.connection === null || event.stream.connection.connectionId === session.connection.connectionId) {
+      event.preventDefault();
+    }
+  });
+
+
+  function callback(error) {
+    // If the connection is successful, publish to the session
+    if (error) {
+      handleError(error);
+    } else {
+      session.publish(publisher, handleError);
+    }
+  }
+
+  // Connect to the session
+  session.connect(token, callback);
+
+  function callOrHangup() {
```

Comment:
removed call. This also made the `publisher streamDestroyed` listener redundant (since we do not plan on destroying the stream ourselves).

Since the subscriber can be `destroyed` without dispatching a `disconnected` event first, I assigned the same listener to both.

Reply:
Done, since we removed the Call button there was also no need for the `publisher streamDisconnected` listener so I removed it.

Additionally, a subscriber can be destroyed without dispatching a `disconnected` event, so I added an additional listener for that.